### ISSUE TITLE
Undefined index: shutdown

### ISSUE THREAD

**Issue by xelan** [mjphaynes/php-resque#59]
The new remote shutdown feature produces a PHP notice, as the new key `shutdown` does not always exist in the packet array.

```
PHP Notice:  Undefined index: shutdown in /path/to/worker/vendor/mjphaynes/php-resque/src/Resque/Worker.php on line 174
```

Probably just using changing the line from `$worker->shutdown = $packet['shutdown'];` to `$worker->shutdown = isset($packet['shutdown']) ? $packet['shutdown'] : null;` would help.

@francislavoie What do you think?

**Comment by francislavoie** [comment]
Sounds fine.

**Comment by francislavoie** [comment]
I looked over this again, I'm actually confused. The initial packet has this: https://github.com/mjphaynes/php-resque/blob/d61040813cb2d737d28965c688c46688af2cb513/src/Resque/Worker.php#L492

When is it not set? Did you have old workers still running or something?

I think this change is still fine, it won't break anything, but I just don't understand how it got into a situation where it wasn't set

**Comment by xelan** [comment]
I tested the new version in a system which has two application servers and three workers. During the upgrade, I started with one worker, stopped and upgraded it while the rest is still running as usual - so during the upgrade process, there were both "old" and "new" job packets in Redis.

Following SemVer, the upgrade from 2.0 to 2.1 should not cause BC breaks. Thus, I propose to keep the `isset()` check until the latest 2.x version and remove it in 3.0.

The same would apply for other changes to Redis data structures. BC breaks would require a fallback (maybe triggering a E_USER_DEPRECATED?) until 3.0.
